sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -82,24 +82,24 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
 
   private def checkLimitLikeClause(name: String, limitExpr: Expression): Unit = {
     limitExpr match {
-      case e if !e.foldable => failAnalysis(
+      case e if !e.foldable => limitExpr.failAnalysis(
```

Comment:
Should `e`  be used inside `match`?

Reply:
I would use it but the type `Expression` of `limitExpr` is enough to invoke `failAnalysis` here. I can replace `limitExpr` by `e` when I will address other comments.